airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx

```diff
@@ -15,30 +15,28 @@ interface StatusCellControlProps {
   isSyncing?: boolean;
   isManual?: boolean;
   id: string;
-  onSync: (id: string) => void;
 }
 
-const ProgressMessage = styled.div`
-  padding: 7px 0;
-`;
-
 export const StatusCellControl: React.FC<StatusCellControlProps> = ({
   enabled,
   isManual,
   id,
   isSyncing,
-  onSync,
   allowSync,
   hasBreakingChange,
 }) => {
+  const { connections } = useConnectionList();
   const { mutateAsync: enableConnection, isLoading } = useEnableConnection();
-  const [{ loading }, OnLaunch] = useAsyncFn(
-    async (event: React.SyntheticEvent) => {
-      event.stopPropagation();
-      onSync(id);
-    },
-    [id]
-  );
+  const { mutateAsync: syncConnection, isLoading: isSyncStarting } = useSyncConnection();
+
+  const onRunManualSync = (event: React.SyntheticEvent) => {
+    event.stopPropagation();
+
+    const connection = connections.find((c) => c.connectionId === id);
+    if (connection) {
+      syncConnection(connection);
+    }
+  };
```

Comment:
This function was basically constructed in three places before (`SourceConnectionTable`, `DestinationConnectionTable` and `ConnectionTable`) and drilled down as prop called `onSync`. Instead we can just create the function here and remove all that intermediate code.